lib/Amazon/events/amazon_event_monitor.rb

```diff
@@ -155,10 +153,24 @@ def parse_instance_state_change(event)
   end
 
   def sns
-    @sns ||= AmazonConnection.sns(@aws_access_key_id, @aws_secret_access_key, @aws_region)
+    @sns ||= aws_connect(:SNS)
   end
 
   def sqs
-    @sqs ||= AmazonConnection.sqs(@aws_access_key_id, @aws_secret_access_key, @aws_region)
+    @sqs ||= aws_connect(:SQS)
+  end
+
+  def aws_connect(service)
+    require 'aws-sdk'
+
+    AWS.const_get(service).new(
+      :access_key_id     => @aws_access_key_id,
+      :secret_access_key => @aws_secret_access_key,
+      :region            => @aws_region,
+
+      :logger            => $aws_log,
+      :log_level         => :debug,
+      :log_formatter     => AWS::Core::LogFormatter.new(AWS::Core::LogFormatter.default.pattern.chomp),
+    )
```

Comment:
Not really; I just flattened AmazonConnection into its three callers. As the duplication wasn't egregious, I didn't bother rearranging this to share the connection with its EMS -- just because I figure that'll happen naturally during Other Upcoming Movements of these classes.